src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -19,13 +27,26 @@ DlgPrefWaveform::DlgPrefWaveform(
     setupUi(this);
 
     // Waveform overview init
-    waveformOverviewComboBox->addItem(tr("Filtered")); // "0"
-    waveformOverviewComboBox->addItem(tr("HSV")); // "1"
-    waveformOverviewComboBox->addItem(tr("RGB")); // "2"
-    m_pTypeControl = std::make_unique<ControlObject>(
-            ConfigKey(QStringLiteral("[Waveform]"),
-                    QStringLiteral("WaveformOverviewType")));
+    waveformOverviewComboBox->addItem(
+            tr("Filtered"), QVariant::fromValue(WOverview::Type::Filtered));
```

Comment:
Is `QVariant::fromValue` needed here? I'm not sure, but the `Q_ENUM` declaration of `WOverview::Type` might declare it as a meta type already. Otherwise, perhaps you can use `Q_DECLARE_METATYPE` explicitly?

Reply:
Unfortunately, neither Q_ENUM nor Q_DECLARE_METATYPE do the trick here after moving the enum class inside the WOverview class, see the convo above https://github.com/mixxxdj/mixxx/pull/13370#discussion_r1641529210 :man_shrugging:

Reply:
Oops my bad, I didn't see that conversation!